### PR TITLE
add product-service

### PR DESCRIPTION
**Pull request by rshuvalov**
Main:
- The getProductsList OR getProductsById lambda function returns a correct response
- The getProductsById AND getProductsList lambda functions return a correct response code 
- Your own Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend

LInk to FE MR - [https://github.com/roman-shuvalov/nodejs-aws-fe/pull/2/files](https://github.com/roman-shuvalov/nodejs-aws-fe/pull/2/files)
Link to FE cloudfront: [https://d2vdbjor2dbwpy.cloudfront.net/](https://d2vdbjor2dbwpy.cloudfront.net/)

Additional scope:
- Async/await is used
- ES6 modules are used
- Webpack is configured
- SWAGGER documentation file is created ([product-service/resources/openapi.yml](https://app.swaggerhub.com/apis/roman-shuvalov/product-service/v1.0.0#/))
- Lambda handlers are covered by basic UNIT tests (used JEST)
- Lambda handlers (getProductsList, getProductsById) code is written in separate files
- Main error scenarious are handled by API: 200, 404, 500

Link to product-service API - [https://x4t5oybu59.execute-api.eu-west-1.amazonaws.com/dev](https://x4t5oybu59.execute-api.eu-west-1.amazonaws.com/dev)

### CODE REVIEW

File: product-service/resources/openapi.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+openapi: 3.0.3
```
Comment (lyolka92) [inline product-service/resources/openapi.yml:1]: Насколько я понимаю, ты руками собирал этот файл.
Есть пакет, который по разделу documentation из серверлесс кофига сам генерит документацию апи - https://www.npmjs.com/package/@conqa/serverless-openapi-documentation#models

File: product-service/test/handlers/get-products-list.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+});
+
+it('Should invoke functions with right params', async () => {
+  productRepository.findProducts.mockResolvedValue(products);
+
+  await getProductsList();
+
+  expect(productRepository.findProducts).toHaveBeenCalled();
+  expect(response.ok).toHaveBeenCalled();
+  expect(response.ok.mock.calls[0][0]).toStrictEqual(products);
```
Comment (lyolka92) [inline product-service/test/handlers/get-products-list.test.js:29]: Если говорить о концепции 3A (arange, act, assert), то тут многовато expect для одного теста.

File: product-service/resources/openapi.yml (hunk 1 above)
Comment (hamanovich) [reply product-service/resources/openapi.yml:1]: Это очень "плохой" пакет, и для продакшн целей его лучше не использовать, он уже устарел и имеет ряд критикал проблем с serverless

File: product-service/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const path = require('path');
+const slsw = require('serverless-webpack');
+
+module.exports = {
+  entry: slsw.lib.entries,
+  mode: 'production',
```
Comment (hamanovich) [inline product-service/webpack.config.js:6]: it's good to distinguish modes: slsw.lib.webpack.isLocal ? 'development' : 'production'

File: product-service/src/services/response.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const response = (data) => {
+  return {
```
Comment (hamanovich) [inline product-service/src/services/response.js:2]: here and below, if function returns the result only, you can omit 'return' keyword and just return the object

**Pull request merged**